scripts/template-updater.ts

```diff
@@ -0,0 +1,197 @@
+import fs from 'fs';
+import path from 'path';
+import semver from 'semver';
+import { err, JSONable, log } from './lib-concourse';
+
+function readFile(filePath: string): string {
+    log(`reading content from file: ${filePath}`);
+    if (fs.existsSync(filePath)) {
+        const stat = fs.statSync(filePath);
+        if (stat.isFile()) {
+            log(`file found: ${filePath}`);
+            const buffer = fs.readFileSync(filePath);
+            return buffer.toString();
+        } else {
+            log(`not a file: ${filePath}`);
+        }
+    } else {
+        log(`file not found: ${filePath}`);
+    }
+    log(`unable to read content from file: ${filePath}`);
+    return null;
+}
+
+function readJSONTemplate(filePath: string): JSONable {
+    try {
+        const content = readFile(filePath);
+        return content !== null && JSON.parse(content);
```

Comment:
`JSON.parse(fs.readFileSync(filePath).toString())`

Seems like you're just adding issues by returning null in a made up function instead of just handling errors here..

Reply:
changed to throw errors